src/domain/opendatafabric/src/identity/dataset_identity.rs

```diff
@@ -221,6 +222,10 @@ impl DatasetName {
     pub fn into_local_ref(self) -> DatasetRef {
         DatasetRef::Alias(DatasetAlias::new(None, self))
     }
+
+    pub fn lowercase_eq(&self, other: &Self) -> bool {
+        self.to_lowercase() == other.to_lowercase()
```

Comment:
1) There is also [to_ascii_lowercase](https://doc.rust-lang.org/std/primitive.str.html#method.to_ascii_lowercase) that is probably much more performant than general unicode version.

2) Even `to_ascii_lowercase` returns a `String` i.e. it allocates per every call. I think this is too much overhead for our case, since we expect that most strings will already be lowercase. IMO it's a design error in std lib that `to_ascii_lowercase` does not return `Cow<str>`.

Reply:
True, even if its return Strign it still keeps less characters and it will be more efficient